packages/hash/frontend/src/pages/[shortname]/entities/[entity-uuid].page/entity-editor/types-section.tsx

```diff
@@ -1,59 +1,119 @@
-import { faAsterisk } from "@fortawesome/free-solid-svg-icons";
-import { FontAwesomeIcon } from "@hashintel/hash-design-system";
+import { extractBaseUri, extractVersion } from "@blockprotocol/type-system";
+import { EntityId } from "@hashintel/hash-shared/types";
+import { versionedUriFromComponents } from "@hashintel/hash-subgraph/src/shared/type-system-patch";
 import { getEntityTypeById } from "@hashintel/hash-subgraph/src/stdlib/element/entity-type";
 import { getRoots } from "@hashintel/hash-subgraph/src/stdlib/roots";
-import { Box, Typography } from "@mui/material";
+import { Box } from "@mui/material";
+import { useEffect, useState } from "react";
 
+import { useBlockProtocolUpdateEntity } from "../../../../../components/hooks/block-protocol-functions/knowledge/use-block-protocol-update-entity";
+import { useBlockProtocolAggregateEntityTypes } from "../../../../../components/hooks/block-protocol-functions/ontology/use-block-protocol-aggregate-entity-types";
 import { SectionWrapper } from "../../../shared/section-wrapper";
-import { WhiteCard } from "../../../shared/white-card";
 import { useEntityEditor } from "./entity-editor-context";
-
-interface TypeCardProps {
-  url: string;
-  title: string;
-}
-
-const TypeCard = ({ url, title }: TypeCardProps) => {
-  return (
-    <WhiteCard href={url}>
-      <Box
-        sx={{
-          display: "flex",
-          alignItems: "center",
-          px: 1.5,
-          py: 1.25,
-          gap: 1.25,
-          color: ({ palette }) => palette.black,
-        }}
-      >
-        <FontAwesomeIcon icon={faAsterisk} />
-        <Typography variant="smallTextLabels" fontWeight={600}>
-          {title}
-        </Typography>
-      </Box>
-    </WhiteCard>
-  );
-};
+import { EntityVersionUpdateModal } from "./types-section/entity-version-update-moda";
+import { TypeCard } from "./types-section/type-card";
 
 export const TypesSection = () => {
-  const { entitySubgraph } = useEntityEditor();
+  const { entitySubgraph, refetch } = useEntityEditor();
 
   const entity = getRoots(entitySubgraph)[0]!;
+  const { updateEntity } = useBlockProtocolUpdateEntity();
+  const {
+    metadata: { editionId, entityTypeId },
+    properties,
+  } = entity;
+
+  const { aggregateEntityTypes } = useBlockProtocolAggregateEntityTypes();
+  const [newVersion, setNewVersion] = useState<number>();
+  const [updateModalOpen, setUpdateModalOpen] = useState(false);
+  const [updatingVersion, setUpdatingVersion] = useState(false);
+
+  useEffect(() => {
+    const init = async () => {
+      const res = await aggregateEntityTypes({ data: {} });
+
+      const entityTypeWithSameBaseId = res.data?.roots.find(
+        (root) => root.baseId === extractBaseUri(entityTypeId),
```

Comment:
I'd `extract` outside of the `find` call, it'll be slightly more efficient. (Otherwise you're extracting on every single root)